### ISSUE TITLE
turtlebot_gazebo not connecting /gazebo and turtlebot topics (odom, mobile_base)

### ISSUE THREAD

**Issue by salfityj** [erlerobot/gym-gazebo#127]
Some preliminaries:
Ubuntu 14.04
ROS Kinetic
Python 2.7
Gazebo 8

TL;DR - Turtlebot isn't able to accept move commands because `/gazebo` and `/mobile_base` are not communicating properly.

Went through all the installation steps and I'm seeing a major issue. 
First off - to get this to build - I had to modify the `gazebo.repos` file
```
ar_track_alvar:
   type: git
   url: https://github.com/sniekum/ar_track_alvar
   version: kinetic-devel
...
ecl_core:
     type: git
     url: https://github.com/stonier/ecl_core
     version: release/0.61-indigo-kinetic
 ecl_lite:
    type: git
    url: https://github.com/stonier/ecl_lite
    version: release/0.61-indigo-kinetic
ecl_navigation:
    type: git
    url: https://github.com/stonier/ecl_navigation
    version: release/0.60-indigo-kinetic
ecl_tools:
    type: git
    url: https://github.com/stonier/ecl_tools
    version: release/0.61-indigo-kinetic
```
Everything built well after running `bash setup_kinetic.bash`

In all the examples in `/gym-gazebo/examples/turtlebot/`, when launched, the turtlebot is spawned in gazebo but there the turtlebot doesn't move. Upon diving deeper, I've found that the turtlebot (kobuki base) is not hooked up to gazebo properly. 

![rqt_graphpython](https://user-images.githubusercontent.com/18411911/38531066-03c0f7dc-3c23-11e8-8377-797b736e434a.JPG)

From `rqt_graph` output, you can see that the `/mobile_base` is not communicating well with `/gazebo`, meaning that the turtlebot won't be able to accept commands to move.

Examining an even simpler example, I'm trying to run the example `roslaunch turtlebot_gazebo turtlebot_world.launch` When doing a 1:1 comparison with another machine with Gazebo7, I'm noticing the `/gazebo` node not publishing important topics including `/odom` and some `/mobile_base` topics among others.

Here it is with Gazebo8
![rqt_graph](https://user-images.githubusercontent.com/18411911/38531484-da46d758-3c24-11e8-931a-977a770c4377.JPG)

Here it is with Gazebo7
![rqt_graphtb](https://user-images.githubusercontent.com/18411911/38531455-b84a49d2-3c24-11e8-9017-b33b44bd4798.JPG)

The difference is that the `/mobile_base` node is not communicating with `/gazebo`
I'm boiling this down to the file `/turtlebot_gazebo/launch/includes/kobuki.launch.xml` not setting things up like it should, but I can't find out what the problem is.

Has anyone got this combo (ubuntu 16.04, ROS Kinetic, Gazebo 8, Python 2.7) and did you go through any special steps? 

Thanks

**Comment by JaxDraxler** [comment]
Hello !!
I don't think that ROS Kinetic works with Ubuntu 14.04

**Comment by salfityj** [comment]
@JaxDraxler typo - thanks for catching my mistake !
Ubuntu 16*

**Comment by JaxDraxler** [comment]
ar_track_alvar:
   type: git
   url: https://github.com/sniekum/ar_track_alvar
   version: kinetic-devel
...
ecl_core:
     type: git
     url: https://github.com/stonier/ecl_core
     version: release/0.61-indigo-kinetic
 ecl_lite:
    type: git
    url: https://github.com/stonier/ecl_lite
    version: release/0.61-indigo-kinetic
ecl_navigation:
    type: git
    url: https://github.com/stonier/ecl_navigation
    version: devel
ecl_tools:
    type: git
    url: https://github.com/stonier/ecl_tools
    version: release/0.61-indigo-kinetic

I guess you should leae the ecl_navigation version to devel. Because when i've changed it to release it raised an error message.
